### DISCUSSION TITLE
آموزش: چگونه مشکلات اسکن را عیب یابی کنیم. (نسخه ویندوزی)

### DISCUSSION THREAD

**Discussion by MortezaBashsiz** [MortezaBashsiz/CFScanner#331] [HowTo]
## عیب یابی (Diagnose) چیست؟
گاهی هیچ نتیجه ای از اسکن کردن بدست نمی آوریم و نیاز داریم تا تمام مراحل اسکن را عیب یابی نماییم تا متوجه شویم که مشکل از کجا می تواند باشد. در نسخه ویندوزی سه مرحله اصلی اسکن که شامل موارد زیر می باشد را میتوانید عیب یابی کرده تا هر خطایی که در این مراحل رخ میدهد را مشاهده نمایید:

1. Fronting 
2. v2ray.exe Execution
3. Download

توجه: مرحله دوم که اجرای v2ray با استفاده از کانفیگ داخلی برنامه یا کانفیگ شخصی شما می باشد همیشه باید بدون خطا و همراه با نتیجه SUCCESS باشد.

در نسخه ویندوزی شما میتوانید عملیات اسکن خود را به دو طریق کلی عیب یابی نمایید. 

1. اجرای برنامه در حالت Debug Mode
2. استفاده از گزینه Diagnose در بخش ها مختلف برنامه

## اجرا در حالت Debug Mode
برای اینکه برنامه را در این حالت اجرا نمایید به فولدر برنامه بروید و در آنجا یک فایل با نام `enable-debug` ایجاد نمایید و سپس برنامه را اجرا نمایید.
 برنامه در حالت دیباگ اجرا میشود و باید بتوانید این پیام را در آن ببینید:
![image](https://user-images.githubusercontent.com/126115050/228255210-6fad7089-0ad8-4483-8901-f6bd527859b3.png)

در این حالت گزینه Concurrent غیر فعال و بر روی عدد یک تنظیم می شود.
بعد از اینکه برنامه در این حالت اجرا شد عملیات اسکن را آغاز کنید. اطلاعات مربوط به عیب یابی اسکن به طور خودکار در فایلی به نام`debug.txt` در فولدر برنامه ذخیره میگردد.
با مطالعه اطلاعات داخل این فایل میتوانید متوجه بشوید در چه بخشی ایراد رخ داده است. همچنین میتوانید این فایل را برای ما در بخش Issues ارسال کنید تا مشکل شما را بررسی نماییم.

## استفاده از گزینه Diagnose در داخل برنامه
در بخش های مختلف برنامه گزینه Diagnose قرار داده شده است که با استفاده از آن شما میتوانید بر روی یک IP خاص فرایند اسکن را اجرا نموده و بر روی کل مراحل آن عیب یابی نمایید. بدین شکل که بعد از اجرای این عمل یک پنجره به مانند زیر باز شده و تمام مراحل اجرای اسکن را گزارش میکند و اگر در بخشی خطایی رخ داد باشد آن را به شما نمایش میدهد:
![image](https://user-images.githubusercontent.com/126115050/228258219-9493a0dd-cc1f-4feb-864f-17101b32e97d.png)

در انتها یک گزارش از سه بخشی اصلی اسکن به شما میدهد که هر کدام با موفقیت بوده اند یا خیر. بدین شکل به راحتی میتوانید محل رخداد خطا را پیدا کنید.

### از کجا این عیب یابی را اجرا کنیم؟
این گزینه در حال حاظر از سه نقطه قابل اجرا می باشد:
1. از منوی Tools بخش Diagnose گزینه `With a random IP address`
    در این حالت یک IP رندوم از بین IP های کلودفلر انتخاب میشود و با آن عیب یابی انجام میشود. احتمال اینکه این IP کار نکند بالا است اما به هر حال برای انجام عیب یابی سریع مفید می باشد. ضمنا با کلید `CTRL + R` میتوانید این گزینه را اجرا نمایید.
2. منوی Tools بخش Diagnose گزینه `With specific IP address...`
    با استفاده از این گزینه یک فرم باز شده و از شما میخواد تا IP مورد نظر خود را وارد نمایید تا با استفاده از آن عیب یابی انجام شود.
5. در تب Scan Results بر روی لیست نتایج راست کلیک کرده و گزینه `Diagnose with this IP address` را انتخاب کنید:
![image](https://user-images.githubusercontent.com/126115050/228263383-bfd70eee-f8b0-4dfe-93a1-f9d94d00d6d6.png)

### نکات

- هنگامیکه کانفیگ شخصی خود را به برنامه اضافه می نمایید، همان ابتدا برنامه از شما می پرسد که آیا میخواهید عملیات عیب یابی را با استفاده از این کانفیگ انجام دهید؟ بدین صورت در همان ابتدا میتوانید مطمئن شوید که کانفیگ شما با برنامه سازگار است یا خیر.
![image](https://user-images.githubusercontent.com/126115050/228264613-e759cc4c-792b-4d77-808f-c3244ec6aa5b.png)
توجه کنید که در نتایج پایانی **حتما** نتیجه **v2ray.exe Execution** باید **SUCCESS** باشد و اگر نباشد یعنی کانفیگ شما مشکلی دارد که باید رفع گردد. برای رفع مشکل با دقت پیام خطایی که در این بخش نمایش میدهد را بخوانید.
- تنظیمات برنامه مانند سرعت دانلود و کانفیگ و تایم اوت در عملیات عیب یابی لحاظ میشود و لذا شما میتوانید عیب یابی را با استفاده از هر کانفیگ و هر سرعت دانلودی انجام دهید.
- اگر خطایی در مورد عدم امکان Listen کردن مشاهده کردید احتمالا در کانفیگ شخصی شما port خاصی در بخش inbound فعال می باشد که باید آن را از کانفیگ خود حذف نمایید. و یا اینکه یک پروسه v2ray از قبل در حافطه سیستم عامل باز مانده است که با استفاده از Task Manager ویندوز تمام پروسه های v2ray.exe را End Task نمایید تا مشکل حل شود.
-  FAILED شدن بخش Fronting و Download الزاما به معنای خطا نمی باشد و فقط میتواند نشانه کار نکردن IP مورد نظری که عیب یابی را روی آن انجام داده اید باشد.
- در نهایت اگر متوحه مشکل نشدید از بخش Issues مشکل خود را به همراه اطلاعات عیب یابی و توضیحات کامل مطرح نمایید.

مرتبط: [چگونه کانفیگ شخصی را به برنامه اضافه نماییم](https://github.com/MortezaBashsiz/CFScanner/discussions/210)

**Comment by sina1100** [comment]
سلام آقا مرتضی من زیاد وارد نبست و برنامه رو دان کردم در ویندوز ، برای شروع و پیدا کردن ای پی تمیز 
از کجا شروع کنم و چه دکمه ایی را کلیک کنم ...در ضمن ممنون بابت همه زحماتت عزیز و دمت گرم

**Comment by p3im4n** [comment]
درود و ادب
آقای مرتضی....آقای سگارو.....اول ممنون بابت همه چیز .
کانفیگ من  vmess+gprc+tls  هست.
من آی پی این ساب "mci.ircf.space"  رو تو رنج خودش نمیتونم به دست بیارم هیچ جوره !!
ولی تو همین رنج آی پی هایی به دست میاد که سرعت خوبی ندارن.

تایم اوت رو بالا بردم....
سرعت رو پایین آووردم....
تعداد ترد ها....
ولی بازم نمیتونم تو نتیجه ها ببنیمش!!

ولی آی پی ش رو به صورت تکی که تست میکنم نتیجه خوبی میده !!!

این مفهومش برا من اینه که قطعا آی پی های دیگری هم هستن که سرعت خوبی دارن ولی در تست های من پیدا نمیشن!
ممنون میشم بم کمک کنید
شاید مشکل بقیه هم باشه ولی متوجه ش نمیشن

ممنون

**Comment by boyluti** [comment]
سلام اقا مرتضی. . اسکنر نسخه ویندوز به نظر مشکلی پیدا کرده. بی زحمت چک کنین. خبرشو داخل کانال تلگرام بدین. نسخه ویندوز رو بیشتر اپدیت کنین.  زحمت نت ایران رو شما ب تنهایی کشیدین. واقعا دمتون گرم. مرسیییی. منتظرییم.  تشکرر فراوان

**Comment by boyluti** [comment]
![errrrrrr](https://github.com/MortezaBashsiz/CFScanner/assets/134806450/3c8f303e-2f2d-49c8-8c4e-9e9cf4af8630)
  این درسته ؟   چرا اول میزنه هنشک ساکسس  اما بعدش میزنه فیلور

**Comment by boyluti** [comment]
اسکنر از کار افتاده. انگار اختلالی انداختن روی اسکنر کلاد فلر. هیچی نمیاره ای پی تمیز.  کمک کنین

**Comment by Vito-cactus** [comment]
کلا تمام اسکنرهای وینددوز و اندروید و پایتون اختلال دارن، بعد از 3 روز سرچ کلا تونستم یک ip  پیدا کنم

**Comment by hossein2** [comment]
{
  "tag": "VLESS + GRPC + TLS",
  "listen": "0.0.0.0",
  "port": "PORTPORT",
  "port": 2083,
  "protocol": "vless",
  "settings": {
      "clients": [],
      "decryption": "none"
    },
  "streamSettings": {
    "network": "grpc",
    "grpcSettings": {
        "serviceName": "2083vless"
      },
    "security": "tls",
    "tlsSettings": {
      "serverName": "ادرس ای پی سرور",
      "address": "IP.IP.IP.IP", 
      "certificates": [
        {
          "ocspStapling": 3600,
          "certificateFile": "/var/lib/marzban/certs/fullchain.pem",
          "keyFile": "/var/lib/marzban/certs/key.pem"
        }
      ],
      "minVersion": "1.2",
      "cipherSuites": "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256:TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"
    }
  },
  "sniffing": {
    "enabled": true,
    "destOverride": ["http", "tls"]
  }
}
سلام من هرچی اسکن میکنم هیچی پیدا نمیکنه میتونی ببینی کجای تگ اشتباه گذاشتم با اینکه SUCCESS  میده
